bundles/org.openhab.binding.linky/src/main/java/org/openhab/binding/linky/internal/console/LinkyCommandExtension.java

```diff
@@ -0,0 +1,134 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.linky.internal.console;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.Arrays;
+import java.util.List;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingRegistry;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.eclipse.smarthome.io.console.Console;
+import org.eclipse.smarthome.io.console.extensions.AbstractConsoleCommandExtension;
+import org.eclipse.smarthome.io.console.extensions.ConsoleCommandExtension;
+import org.openhab.binding.linky.internal.handler.LinkyHandler;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * The {@link LinkyCommandExtension} is responsible for handling console commands
+ *
+ * @author Laurent Garnier - Initial contribution
+ */
+
+@NonNullByDefault
+@Component(service = ConsoleCommandExtension.class)
+public class LinkyCommandExtension extends AbstractConsoleCommandExtension {
+
+    private static final String REPORT = "report";
+
+    private final ThingRegistry thingRegistry;
+
+    @Activate
+    public LinkyCommandExtension(final @Reference ThingRegistry thingRegistry) {
+        super("linky", "Interact with the Linky binding.");
+        this.thingRegistry = thingRegistry;
+    }
+
+    @Override
+    public void execute(String[] args, Console console) {
+        if (args.length >= 2) {
+            LinkyHandler handler = null;
+            try {
+                ThingUID thingUID = new ThingUID(args[0]);
+                Thing thing = thingRegistry.get(thingUID);
+                if ((thing != null) && (thing.getHandler() != null) && (thing.getHandler() instanceof LinkyHandler)) {
+                    handler = (LinkyHandler) thing.getHandler();
+                }
+            } catch (Exception e) {
```

Comment:
What exception are you expecting here?

Reply:
At least, I see that `new ThingUID(args[0])` could throw `IllegalArgumentException`.
I don't think `thingRegistry.get(thingUID)` can throw any exception.

In fact, I copied this pattern from my Powermax binding.

Reply:
Changed to IllegalArgumentException